network/proto/v1/network.proto

```diff
@@ -0,0 +1,370 @@
+syntax = "proto3";
+package messari.network.v1;
+
+message BigInt {
+  bytes bytes = 1;
+}
+
+message Network {
+    // Name of the network being indexed
+    string id = 1;
+  
+    // The verison of the schema
+    string schema_version = 2;
+  
+    // The version of the subgraph
+    string subgraph_version = 3;
+  
+    // The version of the methodology
+    string methodology_version = 4;
+  
+    // #### Quantitative Data ####
+  
+    // Number of cumulative unique authors (see type Author for explanation)
+    BigInt cumulative_unique_authors = 5;
+  
+    // The total number of blocks in the blockchain
+    uint64 block_height = 6;
+  
+    // Total difficulty to reach the current block
+    BigInt cumulative_difficulty = 7;
+    
+    // Total gas used
+    BigInt cumulative_gas_used = 8;
+  
+    // Current gas limit
+    uint64 gas_limit = 9;
+  
+    // Total burnt fees on the network in native asset
+    BigInt cumulative_burnt_fees = 10;
+  
+    // Total rewards emitted on the network in native asset
+    BigInt cumulative_rewards = 11;
+  
+    // Total number of transactions that have occured on this network
+    BigInt cumulative_transactions = 12;
+  
+    // Total size of the network in native units (e.g. Bytes in Arweave and Ethereum)
+    BigInt cumulative_size = 13;
+  
+    // Current supply of the native asset
+    uint64 total_supply = 14;
+  
+    // Statistical analysis of blocks/day from daily snapshots
+    Stats daily_blocks = 15;
+  
+    // #### Snapshots ####
+  
+    // Daily metrics for this blockchain
+    DailySnapshots daily_snapshots = 16;
+  
+    // Hourly metrics for this blockchain
+    HourlySnapshots hourly_snapshots = 17;
+
+    NetworkSnapshot network_snapshot = 18;
```

Comment:
Adding `NetworkSnapshot` based on discussion on https://www.notion.so/messari/Subgraph-Schema-feedback-bc070812d29e4aea9886e4d7201ef206

These `Stats` are in addition to the cumulative values in the Network schema.

Q: Is this the way we are thinking about the `Network` schema?

Reply:
@dmelotik Thoughts on the schema? I also added a note below discussing some of the limitations we may encounter with substreams in terms of how we handle calculations, e.g. keeping a record of values in the `Stats` schema.